### PR TITLE
Fix dual time restarts with UNST_CFL_NUMBER != 0

### PR DESCRIPTION
**Pull request by pcarruscag**
## Proposed Changes
Fix this issue https://www.cfd-online.com/Forums/su2/235718-unsteady-2nd-order-dual-time-step-diverges-when-restarted-steady-solution.html
Plus other cleanups

### CODE REVIEW

File: Common/src/CConfig.cpp [hunk 1]
```diff
@@ -2862,34 +2856,65 @@ void CConfig::SetConfig_Parsing(istream& config_buffer){
...
-          if (!option_name.compare("WRT_SOL_FREQ"))
+          else if (!option_name.compare("WRT_SOL_FREQ"))
             newString.append("WRT_SOL_FREQ is deprecated. Use OUTPUT_WRT_FREQ instead.\n\n");
-          if (!option_name.compare("WRT_SOL_FREQ_DUALTIME"))
+          else if (!option_name.compare("WRT_SOL_FREQ_DUALTIME"))
             newString.append("WRT_SOL_FREQ_DUALTIME is deprecated. Use OUTPUT_WRT_FREQ instead.\n\n");
+          else if (!option_name.compare("UNST_RESTART_ITER"))
+            newString.append("UNST_RESTART_ITER is deprecated. Use RESTART_ITER instead.\n\n");
+          else if (!option_name.compare("DYN_RESTART_ITER"))
+            newString.append("DYN_RESTART_ITER is deprecated. Use RESTART_ITER instead.\n\n");
```
Comment (pcarruscag) [inline Common/src/CConfig.cpp:2886]: Another two obsolete options.

File: SU2_CFD/include/solvers/CFVMFlowSolverBase.hpp [hunk 2]
```diff
@@ -524,9 +524,10 @@ class CFVMFlowSolverBase : public CSolver {
 
     }
 
-    /*--- Recompute the unsteady time step for the dual time strategy if the unsteady CFL is diferent from 0. ---*/
+    /*--- Recompute the unsteady time step for the dual time strategy if the unsteady CFL is diferent from 0.
+     * This is only done once because in dual time the time step cannot be variable. ---*/
 
-    if ((dual_time) && (Iteration == 0) && (config->GetUnst_CFL() != 0.0) && (iMesh == MESH_0)) {
+    if (dual_time && (Iteration == config->GetRestart_Iter()) && (config->GetUnst_CFL() != 0.0) && (iMesh == MESH_0)) {
```
Comment (pcarruscag) [inline SU2_CFD/include/solvers/CFVMFlowSolverBase.hpp:530]: This is the fix, the rest was cleanup.

**Review by TobiKattmann — APPROVED** [review]
thanks for fixing and cleaning 💐

File: Common/src/CConfig.cpp [hunk 3]
```diff
@@ -2862,34 +2856,65 @@ void CConfig::SetConfig_Parsing(istream& config_buffer){
...
           // This option is deprecated. After a grace period until 7.2.0 the usage warning should become an error.
-          /*if (!option_name.compare("CONV_CRITERIA"))
+          /*else if (!option_name.compare("CONV_CRITERIA"))
             newString.append(string("CONV_CRITERIA is deprecated. SU2 will choose the criteria automatically based on the CONV_FIELD.\n") +
                              string("RESIDUAL for any RMS_* BGS_* value. CAUCHY for coefficients like DRAG etc.\n\n"));*/
+          else {
+            auto countMatchChars = [&option_name](const string& candidate) {
+              const size_t sz1 = option_name.size(), sz2 = candidate.size();
+              size_t nMatch = 0;
```
[the remark this answers is not in the document]
Comment (pcarruscag) [reply Common/src/CConfig.cpp]: Yep I put it in the other PR #1269 after you mentioned it in the meeting.
Thanks for the review.

**Pull request merged**